frontend/pkg/frontend/frontend.go

```diff
@@ -167,7 +165,15 @@ func (f *Frontend) Join() {
 	<-f.done
 }
 
-func (f *Frontend) CheckReady() bool {
+func (f *Frontend) CheckReady(ctx context.Context) bool {
+	// Verify the DB is available and accessible
+	result, err := f.dbClient.DBConnectionTest(ctx)
+	if err != nil {
+		f.logger.Error(fmt.Sprintf("Database test failed to fetch properties: %v", err))
+		return false
+	}
```

Comment:
Would it make more sense for DBConnectionTest() to just return an error or boolean instead of a string and a boolean? Will that string be never useful?

Reply:
This is a good point, sorry for merging prematurely.

Looking over what the string return is for, I'd suggest keep the return signature but change `"No database configured, skipping"` to an empty string.  That would let us more easily distinguish a live database from no database, which _might_ be useful in the future.

Reply:
@mbarnes No worries 🙂 

For the case of differentiating between 
- not having the database live vs 
- not having the DB configured (or any other case)
I would rather use 2 centinel errors that we can compare against, as this string that is returned as part of the DB test call has no specific meaning. 

As I see it, the interface provides a contract and in that case, the contract includes a field that can mean or be used for anything which can lead to strange uses of it. Just my two-cents on this, not super important.

Reply:
https://github.com/Azure/ARO-HCP/pull/162